pallets/xvm/src/lib.rs

```diff
@@ -84,6 +90,11 @@ pub mod pallet {
         /// `CheckedEthereumTransact` implementation.
         type EthereumTransact: CheckedEthereumTransact;
 
+        /// Max weight limit allowed for cross-VM calls. Calls will fail
+        /// if their `ref_time` or `proof_size` exceeds this limit.
+        #[pallet::constant]
+        type MaxWeightLimit: Get<Weight>;
```

Comment:
Sorry if I wasn't clear enough with my original comment.
I was thinking about this (from `xvm` precompiles):
```Rust
        let remaining_gas = handle.remaining_gas();
        let weight_limit = R::GasWeightMapping::gas_to_weight(remaining_gas, true);
        let xvm_context = Context {
            source_vm_id: VmId::Evm,
            weight_limit,
        };
```

It always takes the `remaining_gas()`, but perhaps user specified a custom limit.
`PrecompileHandle`'s `call` function allows this:
https://github.com/rust-blockchain/evm/blob/b7b82c7e1fc57b7449d6dfa6826600de37cc1e65/src/executor/stack/precompile.rs#L40

Reply:
Thanks for the clarification. This makes sense, will update to use the min of remaining and user's setting instead.

Previously I thought you meant the max weight limit setting related to an issue reported before (in the XVM backlog).

Reply:
Updated precompile to respect caller's gas limit setting. For chain extension, I might miss something, but it seems there is no existing API for users to specific gas limit. We'll need to add an extra parameter to XVM call interface.

Reply:
[Here](https://github.com/paritytech/ink-examples/blob/3254a292b5cf13b087f92f2d02dc4bcbe847e71d/call-runtime/lib.rs#L97-L102) is an example of `call_runtime` usage. The function doesn't have to only accept one parameter. I think it's reasonable for them to add a parameter to specify the weight limit, like [`CallBuilder`](https://use.ink/basics/cross-contract-calling/#callbuilder) use case.

Reply:
Yes, it makes sense - caller should have some control over how much weight are they willing to expend on the call, same as any other.

Reply:
> 1. make ink! env support weight v2; 2. add custom weight support to call_runtime interface.

Let's create github issue for this to start discussion in ink! repo

Reply:
> Let's create github issue for this to start discussion in ink! repo

Updated issue #987 to summarize the discussion here. It will only be closed when the above `call_runtime` solution is finished.